Source/core/JSON.h

```diff
@@ -2809,9 +2811,13 @@ namespace Core {
                         offset = PARSE;
                     }
                 }
-                if ((offset == static_cast<uint16_t>(~0)) && (loaded < maxLength)) {
-                    stream[loaded++] = ']';
-                    offset = 0;
+                if (offset == static_cast<uint16_t>(~0)) {
+                    if (loaded < maxLength) {
+                        stream[loaded++] = ']';
+                        offset = 0;
+                    } else {
+                        offset = END_MARKER;
```

Comment:
It seems the unit tests also indicate this causes some regression. As Pierre mentions the offset of 6 now has a special meaning.

Reply:
Ok, @MFransen69